docs/gettingstarted/index.md

```diff
@@ -1,60 +1,119 @@
+
+3. Open your selected IDE:
+    * Forge only explicitly supports development on Eclipse and IntelliJ IDEA, but there are additional run configurations for Visual Studio Code. Regardless, any environment, from Apache NetBeans to Vim / Emacs, can be used.
+    * Eclipse and IntelliJ IDEA's Gradle integration, both installed and enabled by default, will handle the rest of the initial workspace setup on import or open. This includes downloading the necessary packages from Mojang, MinecraftForge, etc. The 'Gradle for Java' plugin is needed for Visual Studio Code to do the same.
+    * Gradle will need to be invoked to re-evaluate the project for almost all changes to its associated files (e.g., `build.gradle`, `settings.gradle`, etc.). Some IDEs come with 'Refresh' buttons to do this; however, it can be done through the terminal via `gradlew build --refresh-dependencies`.
+3. Run configurations can be generated for Eclipse, IntelliJ IDEA, or Visual Studio Code:
+    * **Eclipse**: Run the `genEclipseRuns` task.
+    * **IntelliJ IDEA**: Run the `genIntellijRuns` task. If a "module not specified" error occurs, set the [`ideaModule` property][config] to your 'main' module (typically `${project.name}.main`).
+    * **Visual Studio Code**: Run the `getVSCodeRuns` task.
+    * **Other IDEs**: You can run the configurations directly using `gradle run*` (e.g., `runClient`, `runServer`, `runData`, `runGameTestServer`). These can also be used with the supported IDEs.
 
 Customizing Your Mod Information
 --------------------------------
 
-Edit the `build.gradle` file to customize how your mod is built (the file names, versions, and other things). Most things in the `build.gradle` can be removed or customized there.
+Edit the `build.gradle` file to customize how your mod is built (e.g., file name, artifact version, etc.).
 
 !!! important
-    **Do not** edit the `settings.gradle` unless you know what you are doing. The default text is necessary to get the ForgeGradle plugin.    
+    Do **not** edit the `settings.gradle` unless you know what you are doing. The file specifies the repository that [ForgeGradle] is uploaded to.
+
+### Recommended `build.gradle` Customizations
 
-### Simple `build.gradle` Customizations
+#### Mod Id Replacement
 
-These customizations are highly recommended for all projects.
+Replace all occurrences of `examplemod`, including [`mods.toml` and the main mod file][modfiles] with the mod id of your mod. This also includes changing the name of the file you build by setting `archivesBaseName` (this is typically set to your mod id).
 
-* To change the name of the file you build - edit the value of `archivesBaseName` to suit.
-* To change your "maven coordinates" - edit the value of `group` as well.
-* To change the version number - edit the value of `version`.
-* To update the run configurations - replace all occurrences of `examplemod` to the mod id of your mod.
+```gradle
+// In some build.gradle
+archivesBaseName = 'mymod'
```

Comment:
As a note, I believe Gradle is looking to move away from conventions (which is what allows us to define `archivesBaseName` in the very root of the buildscript, per [`BasePluginConvention`](https://docs.gradle.org/current/dsl/org.gradle.api.plugins.BasePluginConvention.html)) to extensions. 

Specifically, we will likely need to transition this to `base { archivesName = 'mymod' }`. CCing @MinecraftForge/library-maintainers, for future reference.